packages/swing-store/test/test-compression.js

```diff
@@ -0,0 +1,398 @@
+import path from 'path';
+import sqlite3 from 'better-sqlite3';
+import { gunzipSync } from 'zlib';
+import test from 'ava';
+
+import { initSwingStore } from '../src/index.js';
+
+import { tmpDir } from './util.js';
+
+test('when spans are not compressed on rollover', async t => {
+  const [dbDir, cleanup] = await tmpDir('testdb');
+  t.teardown(cleanup);
+  // const dbDir = 't-db';
+
+  const ss = initSwingStore(dbDir);
+  const ts = ss.kernelStorage.transcriptStore;
+
+  const db = sqlite3(path.join(dbDir, 'swingstore.sqlite'));
+  const sqlGetSpans = db.prepare(
+    'SELECT vatID, startPos, endPos, incarnation, isCurrent FROM transcriptSpans ORDER BY startPos',
+  );
+  const sqlGetItems = db.prepare(
+    'SELECT * FROM transcriptItems ORDER BY position',
+  );
+  const sqlGetCompressedSpans = db.prepare(
+    'SELECT vatID, startPos, endPos, incarnation FROM transcriptCompressedSpans ORDER BY startPos',
+  );
+  const vatID = 'v1';
+  let spans;
+  let items;
+  let compressedSpans;
+
+  // to start with, turn automatic compression off
+  ss.debug.setEnableTranscriptRolloverCompression(false);
+
+  // build a DB with four spans (one in an old incarnation, two historical but
+  // current incarnation, only one inUse)
+  ts.initTranscript(vatID);
+  await ss.hostStorage.commit();
+
+  spans = [...sqlGetSpans.iterate()];
```

Comment:
nit: given how many times these three are called, I'd factor them into a helper, and use `cur = helper()` and `t.deepEqual(cur.items, ..`